frontend/src/components/Profile.tsx

```diff
@@ -14,15 +14,17 @@ const Profile: React.FC<ProfileProps> = ({ user, updateUser }) => {
   const [playerTag, setPlayerTag] = useState<string>(user.playerTag);
   const [email, setEmail] = useState<string>(user.email);
   const imageInput = useRef<HTMLInputElement>(null);
-  const [imagePreview, setImagePreview] = useState<string>(user.image);
+  const [imagePreview, setImagePreview] = useState<string>(
+    `${apiBaseUrl}/public/image/${user.image}`
+  );
   const [password, setPassword] = useState<string>('');
   const [modify, setModify] = useState<boolean>(false);
-  const formData = new FormData();
+  const formData = useRef<FormData>(new FormData());
```

Comment:
this would produce a bug of editing the email address and then resetting back to how it was but still being sent to the backend which throws an error, try keeping track of edited fields using an object as a state, whenever a change occurs in the input fields, compare it with the initial value of that input field, if it is the same, remove the field from the edited fields objects, otherwise add it to it. when sending the data inside the submit function, you instantiate a form data object, append the edited attributes to it and use it.